tests/tests/vertex_indices/mod.rs

```diff
@@ -356,20 +337,17 @@ async fn vertex_index_common(ctx: TestingContext) {
         )
         .create_view(&wgpu::TextureViewDescriptor::default());
 
-    let mut tests = Vec::with_capacity(5 * 2 * 2 * 2);
+    let mut tests = Vec::with_capacity(5 * 2 * 2);
     for case in TestCase::ARRAY {
         for id_source in IdSource::ARRAY {
             for draw_call_kind in DrawCallKind::ARRAY {
                 for encoder_kind in EncoderKind::ARRAY {
```

Comment:
thought(non-blocking): These manual `ARRAY` constants are unfortunate. We actually don't need to deal with them; we can use the nice `strum::{IntoEnumIterator, EnumIter}` APIs to `derive` an `Iterator` over the elements of these different `enum`s instead, i.e.:

```diff
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Copy, Clone, strum::EnumIter)] // adds `TestCase::iter()` instead
 enum TestCase {
     // …
 }
 
 impl TestCase {
-    const ARRAY: [Self; 5] = [
-        // …
-    ];
     // …
 }
```

We'll need to add a dependency to `strum` here, but we already have it in our lockfile, so 🤷🏻‍♂️ not worried about dependency bloat. It's a _very_ useful dep. to have.

☝🏻CC folks who participated in introducing this code in <https://github.com/gfx-rs/wgpu/pull/4723> (@cwfitzgerald as author, @teoxoy as reviewer). Also, CC @gfx-rs/wgpu for awareness.

Reply:
Let's move non-blocking feedback in this convo to <https://github.com/gfx-rs/wgpu/pull/5840>.